packages/cozy-client/src/models/contacts.js

```diff
@@ -0,0 +1,128 @@
+import get from 'lodash/get'
+import logger from 'cozy-logger'
+
+export const getPrimaryOrFirst = property => obj =>
+  !obj[property] || obj[property].length === 0
+    ? ''
+    : obj[property].find(property => property.primary) || obj[property][0]
+
+/**
+ * Returns the initials of the contact.
+ *
+ * @param {object} contact - A contact
+ * @returns {string} - the contact's initials
+ */
+export const getInitials = contact => {
+  if (contact.name) {
+    return ['givenName', 'familyName']
+      .map(part => get(contact, ['name', part, 0], ''))
+      .join('')
+      .toUpperCase()
+  }
+
+  const email = getPrimaryEmail(contact)
+  if (email) {
+    return email[0].toUpperCase()
+  }
+
+  logger.warn('Contact has no name and no email.')
+  return ''
+}
+
+/**
+ * Returns the contact's main email
+ *
+ * @param {object} contact - A contact
+ * @returns {string} - The contact's main email
+ */
+export const getPrimaryEmail = contact =>
+  Array.isArray(contact.email)
+    ? getPrimaryOrFirst('email')(contact).address || ''
+    : contact.email
+
+/**
+ * Returns the contact's main cozy
+ *
+ * @param {object} contact - A contact
+ * @returns {string} - The contact's main cozy
+ */
+export const getPrimaryCozy = contact =>
+  Array.isArray(contact.cozy)
+    ? getPrimaryOrFirst('cozy')(contact).url || ''
+    : contact.url
+
+/**
+ * Returns the contact's main phone number
+ *
+ * @param {object} contact - A contact
+ * @returns {string} - The contact's main phone number
+ */
+export const getPrimaryPhone = contact =>
+  getPrimaryOrFirst('phone')(contact).number || ''
+
+/**
+ * Returns the contact's main address
+ *
+ * @param {object} contact - A contact
+ * @returns {string} - The contact's main address
+ */
+export const getPrimaryAddress = contact =>
+  getPrimaryOrFirst('address')(contact).formattedAddress || ''
+
+/**
+ * Returns the contact's fullname
+ *
+ * @param {object} contact - A contact
+ * @returns {string} - The contact's fullname
+ */
+export const getFullname = contact => {
+  if (get(contact, 'fullname')) {
```

Comment:
Not necessarily a problem, just pointing out that this isn't exactly the same as what ytou had previously pushed, since `contact.fullname` was ignored.

Reply:
Oui j'ai vu, celui-ci me paraît mieux du coup